lib/TableGen/Operations.cpp

```diff
@@ -105,19 +108,32 @@ SmallVector<NamedValue> OperationBase::getFullArguments() const {
 }
 
 unsigned OperationBase::getNumFullArguments() const {
+  size_t argSize = m_arguments.size();
+
+  // Variadic argument lists can be of size zero,
+  // so do not add them to the total number of arguments.
+  if (m_hasVariadicArgument && !m_arguments.empty())
+    --argSize;
```

Comment:
Can the condition `m_hasVariadicArgument && m_arguments.empty()` (not the second condition is inverted) be true, and if so, in which cases?

Reply:
Theoretically, it could, if you pass zero arguments and use a varargs binding without any arguments (e. g. `create<Op>({})`). In that case, the expression should evaluate to zero. Good catch. Thanks.

Reply:
From a runtime perspective (e. g. checking against concrete operations), yes. So, that would be a good catch.
From a dialects-compilation perspective, no, as using a variadic argument binding implies having an entry in `m_arguments`.
So, maybe the check to `!m_arguments.empty()` is superfluous, so I'll probably remove it. Thanks.

Reply:
Maybe put it into an assert.

Reply:
I think we can leave it for now.